# HG changeset patch
# User Rik <rik@octave.org>
# Date 1448497234 28800
#      Wed Nov 25 16:20:34 2015 -0800
# Node ID 15eefcabcb316b2011604dd00ac2ebe3e6a16f21
# Parent  2892f62fb37cb9586f98c26948bf8683ddbb9833
doc: Add docstring for internal function __varval__.

* variables.cc (F__varval__): Add one sentence explanation of function.
Recode to place input validation first.

diff --git a/libinterp/corefcn/variables.cc b/libinterp/corefcn/variables.cc
--- a/libinterp/corefcn/variables.cc
+++ b/libinterp/corefcn/variables.cc
@@ -2649,30 +2649,28 @@ void maybe_missing_function_hook (const 
           feval (func_name, octave_value (name));
         }
     }
 }
 
 DEFUN (__varval__, args, ,
        "-*- texinfo -*-\n\
 @deftypefn {Built-in Function} {} __varval__ (@var{name})\n\
-Undocumented internal function.\n\
+Return the value of the variable @var{name} directly from the symbol table.\n\
 @end deftypefn")
 {
   octave_value retval;
 
-  if (args.length () == 1)
-    {
-      std::string name = args(0).xstring_value ("__varval__: first argument must be a variable name");
-
-      retval = symbol_table::varval (args(0).string_value ());
-    }
-  else
+  if (args.length () != 1)
     print_usage ();
 
+  std::string name = args(0).xstring_value ("__varval__: first argument must be a variable name");
+
+  retval = symbol_table::varval (args(0).string_value ());
+
   return retval;
 }
 
 static std::string Vmissing_component_hook;
 
 DEFUN (missing_component_hook, args, nargout,
        "-*- texinfo -*-\n\
 @deftypefn  {Built-in Function} {@var{val} =} missing_component_hook ()\n\
